### PR TITLE
Improve the error message for initializing with wrong stokes shape

### LINKED ISSUE #142: Error message with SkyModel input lengths
When a SkyModel is passed multiple inputs and one is of different length, the error message reads:

`ValueError: UVParameter <pyuvdata.parameter.UVParameter object at 0x7fe0a97efe80> is not expected shape. Parameter shape is (699,), expected shape is (26734,).`

This could be more intuitive to solve if the error indicated which input is of an incorrect length.

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Provide a better error message when SkyModel is initialized with a stokes array that is the incorrect shape (doesn't match the other input shapes)

## Motivation and Context

Fixes #142 

### Other Change Checklist

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyradiosky/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.
- [x] Any new or updated docstrings use the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] I have added tests to cover any changes.
- [ ] My change includes a breaking change
  - [ ] My change includes backwards compatibility and deprecation warnings (if possible).
- [ ] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyradiosky/blob/main/CHANGELOG.md) if appropriate.

### CODE REVIEW

**Review by mkolopanis — COMMENTED** [review]
I have a question in general. Couldn't this whole thing be solved by moving the check before the coherency calculation too? Is there an advantage/disadvantage to either method? obviously check can be turned off and you could still end up with the situation from #142

File: pyradiosky/skymodel.py [hunk 1]
```diff
@@ -15,15 +15,11 @@
...
 from pyuvdata.uvbase import UVBase
 from pyuvdata.parameter import UVParameter
 import pyuvdata.utils as uvutils
-
-try:
-    from pyuvdata.uvbeam.cst_beam import CSTBeam
-except ImportError:  # pragma: no cover
-    # backwards compatility for older pyuvdata versions
-    from pyuvdata.cst_beam import CSTBeam
+from pyuvdata.uvbeam.cst_beam import CSTBeam
```
[the remark this answers is not in the document]
Comment (bhazelton) [reply pyradiosky/skymodel.py:23]: yes, it's there post 2.0 and we require 2.1.3 on pyradiosky, so this is no longer needed.

**Comment by bhazelton** [comment]
We can't call check before we do the coherency calculation because the coherency is required, so the check will error on that being missing. We could make it not required, run check, and then make it required but that seems more complicated (and then the coherency wouldn't get checked).

**Review by mkolopanis — APPROVED** [review]
yeah okay that was the case I was worried about in the second half of my idea. This should work then thanks!!

**Pull request merged**